### ISSUE TITLE
fails to build on 32-bit x86

### ISSUE THREAD

**Issue by nekopsykose** [weidai11/cryptopp#1142]
### Crypto++ Issue Report

* State the operating system and version (Ubutnu 17 x86_64, Windows 7 Professional x64, etc): Alpine linux edge
* State the version of the Crypto++ library (Crypto++ 7.0, Master, etc): 8.7 release
* State how you built the library (Makefile, Cmake, distro, etc): makefile, gcc11

8.7 fails to build `make dynamic libcryptopp.pc` on 32-bit x86:  

```
g++ -DNDEBUG -Os -fomit-frame-pointer -g2 -pthread -pipe -Os -fomit-frame-pointer -c poly1305.cpp
gcm.cpp: In member function 'virtual size_t CryptoPP::GCM_Base::AuthenticateBlocks(const byte*, size_t)':
gcm.cpp:566:13: error: the register '%xmm5' cannot be clobbered in 'asm' for the current target
  566 |             __asm__ __volatile__
      |             ^~~~~~~
gcm.cpp:745:13: error: the register '%xmm1' cannot be clobbered in 'asm' for the current target
  745 |             __asm__ __volatile__
      |             ^~~~~~~
g++ -DNDEBUG -Os -fomit-frame-pointer -g2 -pthread -pipe -Os -fomit-frame-pointer -c polynomi.cpp
make: *** [GNUmakefile:1792: gcm.o] Error 1
```

**Comment by gcsideal** [comment]
Fails to build on Debian x86 as well. [Full build log.](https://buildd.debian.org/status/fetch.php?pkg=libcrypto%2B%2B&arch=i386&ver=8.7.0-1&stamp=1660376866&raw=0)

**Comment by noloader** [comment]
Thanks @nekopsykose and @gcsideal,

This bug has been fixed at [Commit 01a18bdbcb22](https://github.com/weidai11/cryptopp/commit/01a18bdbcb22).

**Comment by noloader** [comment]
Here's a patch if you need it: [x86-build.diff.zip](https://github.com/weidai11/cryptopp/files/9426931/x86-build.diff.zip)
